src/services/upload/uploader.ts

```diff
@@ -50,40 +46,27 @@ export default async function uploader(
             throw Error(CustomError.NO_METADATA);
         }
 
-        if (fileAlreadyInCollection(existingFilesInCollection, metadata)) {
-            addLogLine(`skipped upload for  ${fileNameSize}`);
-            return { fileUploadResult: UPLOAD_RESULT.ALREADY_UPLOADED };
-        }
-
-        const sameFileInOtherCollection = findSameFileInOtherCollection(
-            existingFiles,
-            metadata
-        );
-
-        if (sameFileInOtherCollection) {
-            addLogLine(
-                `same file in other collection found for  ${fileNameSize}`
-            );
-            const resultFile = Object.assign({}, sameFileInOtherCollection);
-            resultFile.collectionID = collection.id;
-            await addToCollection(collection, [resultFile]);
-            return {
-                fileUploadResult: UPLOAD_RESULT.UPLOADED,
-                uploadedFile: resultFile,
-                skipDecryption: true,
-            };
+        const existingFile = findMatchingExistingFile(existingFiles, metadata);
```

Comment:
there could be multiple files which might match the hash.. Here, we are only returning first match. Is that by design? Or will it cause issue?

Reply:
ahh...there is a case where the file is in multiple collection, one the current upload collection (say `x` ) and one another collection (say `y`).

we might get `y` collection's file and we will try to add a symlink for it in collection `x` and that will return a no-op from the server , and we would show the file as uploaded, instead of showing it as ignored already in collection.

Reply:
check for ownership and trashed

Reply:
done this